packages/frontend/src/state/lp/hooks.ts

```diff
@@ -13,6 +13,61 @@ import { ethers } from 'ethers'
 import { useCallback } from 'react'
 import { useGetDebtAmount, useGetTwapSqueethPrice, useGetVault } from '../controller/hooks'
 
+/*** CONSTANTS ***/
+const TICK_SPACE = 60
+const COLLAT_RATIO = 1.5
+
+/*** ACTIONS ***/
+
+// Opening a mint and LP position and depositing
+export const useOpenPositionDeposit = () => {
+  const { squeethPool } = useAtomValue(addressesAtom)
+  const address = useAtomValue(addressAtom)
+  const contract = useAtomValue(controllerHelperHelperContractAtom)
+  const handleTransaction = useHandleTransaction()
+  const getTwapSqueethPrice = useGetTwapSqueethPrice()
+  const getDebtAmount = useGetDebtAmount()
+  const openPositionDeposit = useAppCallback(
+    async (squeethToMint: BigNumber, lowerTickInput: number, upperTickInput: number, onTxConfirmed?: () => void) => {
+      if (!contract || !address) return null
+      
+      const squeethPrice = await getTwapSqueethPrice()
+      const mintWSqueethAmount = fromTokenAmount(squeethToMint, OSQUEETH_DECIMALS)
+      const ethDebt = await getDebtAmount(mintWSqueethAmount)
+
+      const collateralToMint = ethDebt.multipliedBy(COLLAT_RATIO)
+      const collateralToLp = mintWSqueethAmount.multipliedBy(squeethPrice)
+
+      const lowerTick = nearestUsableTick(lowerTickInput, TICK_SPACE)
+      const upperTick = nearestUsableTick(upperTickInput, TICK_SPACE)
+
+      const flashloanWMintDepositNftParams = {
+        wPowerPerpPool: squeethPool,
+        vaultId: 0,
```

Comment:
always vaultId = 0?

Reply:
Ah, fixed!

Reply:
i see this hard coded as vaultId: 0 still